cli/src/commands/git/push.rs

```diff
@@ -279,36 +279,59 @@ pub fn cmd_git_push(
             &tx.base_workspace_helper().revset_parse_context(),
         )?)
         .range(&RevsetExpression::commits(new_heads));
-    for commit in tx
-        .base_workspace_helper()
-        .attach_revset_evaluator(commits_to_push)?
-        .evaluate_to_commits()?
+
     {
-        let commit = commit?;
-        let mut reasons = vec![];
-        if commit.description().is_empty() && !args.allow_empty_description {
-            reasons.push("it has no description");
-        }
-        if commit.author().name.is_empty()
-            || commit.author().name == UserSettings::USER_NAME_PLACEHOLDER
-            || commit.author().email.is_empty()
-            || commit.author().email == UserSettings::USER_EMAIL_PLACEHOLDER
-            || commit.committer().name.is_empty()
-            || commit.committer().name == UserSettings::USER_NAME_PLACEHOLDER
-            || commit.committer().email.is_empty()
-            || commit.committer().email == UserSettings::USER_EMAIL_PLACEHOLDER
+        let workspace_helper = tx.base_workspace_helper();
+
+        let git_settings = command.settings().git_settings();
+        let is_private = if let Some(private_commits) = git_settings.private_commits {
+            let private_commits_revset = revset::parse(
+                private_commits.as_str(),
+                &tx.base_workspace_helper().revset_parse_context(),
+            )?;
+            Some(
+                workspace_helper
+                    .attach_revset_evaluator(private_commits_revset)?
+                    .evaluate()?
+                    .containing_fn(),
+            )
+        } else {
+            None
```

Comment:
nit: could be `Box::new(|_: &CommitId| false)` instead of `Option`

Reply:
Ha, I actually wrote it that way at first but then changed my mind. I'll change it back.